js/components/seismic-eruptions-map.js

```diff
@@ -164,6 +185,21 @@ export default class SeismicEruptionsMap extends PureComponent {
     this.setState({ selectedVolcano: null })
   }
 
+  handleZoom () {
+    if (!config.sizeToFitBounds) {
+      // if we're not using bounds, we use a center point.
+      // ensure map remains with the correct center.
+      const { mark2DViewModified } = this.props
+      const nextViewport = this.refs.map.viewport
+      // using the + and - buttons will not change the center, only double-click and mouse wheel
+      if (nextViewport.center !== INITIAL_CENTER) {
+        nextViewport.center = INITIAL_CENTER
+        this.setState({ mapViewport: nextViewport })
+        mark2DViewModified(false)
+      }
+    }
```

Comment:
I'm not sure if I get the idea here. Why do we need to read the new viewport and overwrite center point?

Reply:
If you scroll in and out with your mouse, your center point will become the area where your mouse is, enabling you to traverse the map without panning it, but just using the zoom controls. There was a method in leaflet that should have prevented this (`setZoomAround`) but I was unable to get it to work in this context, so I re-set the center after a zoom instead as a workaround.

Reply:
This Leaflet wrapper isn't perfect, right. But if there is a Leaflet option, we can definitely use it by accessing Leaflet instance directly. It's even done in this file:
https://github.com/concord-consortium/seismic-explorer/blob/17b8c12488ee8d494b3572f73b2c2ceb4f038eb2/js/components/seismic-eruptions-map.js#L120